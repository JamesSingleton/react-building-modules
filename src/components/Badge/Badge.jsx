import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const SIZES = {
  small: 'small',
  medium: 'medium',
};

const TYPE = {
  solid: 'solid',
  outlined: 'outlined',
};

const SKIN = {
  general: 'general',
  standard: 'standard',
  danger: 'danger',
  success: 'success',
  warning: 'warning',
  premium: 'premium',
};

const StyledBadge = styled.div`
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ${(props) => {
    if (props.skin === SKIN.standard) return props.theme.standard;
    if (props.skin === SKIN.danger) return props.theme.danger;
    if (props.skin === SKIN.success) return props.theme.success;
    if (props.skin === SKIN.warning) return props.theme.warning;
    if (props.skin === SKIN.premium) return props.theme.premium;
    return props.theme.general;
  }};
  padding: ${(props) => {
    if (props.size === SIZES.small) return '0.125rem 0.375rem';
    return '0.3125rem 0.75rem';
  }}
  color: #fff;
  ${(props) => props.type === TYPE.outlined
    && css`
      background-color: #fff;
      color: ${() => {
    if (props.skin === SKIN.standard) return props.theme.standard;
    if (props.skin === SKIN.danger) return props.theme.danger;
    if (props.skin === SKIN.success) return props.theme.success;
    if (props.skin === SKIN.warning) return props.theme.warning;
    if (props.skin === SKIN.premium) return props.theme.premium;
    return props.theme.general;
  }};
    `}
`;

StyledBadge.defaultProps = {
  theme: {
    general: '#32536A',
    standard: '#006fcf',
    danger: '#F25C5E',
    success: '#43a34c',
    warning: '#FDB02B',
    premium: '#9B61B3',
  },
};
const StyledText = styled.span`
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
`;

const Badge = ({
  text, onClick, size, ...rest
}) => (
  <StyledBadge onClick={onClick} size={size} {...rest}>
    <StyledText>{text}</StyledText>
  </StyledBadge>
);

Badge.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  size: PropTypes.oneOf(Object.keys(SIZES)),
  type: PropTypes.oneOf(Object.keys(TYPE)),
  skin: PropTypes.oneOf(Object.keys(SKIN)),
};

Badge.defaultProps = {
  onClick: () => undefined,
  size: SIZES.medium,
  type: TYPE.solid,
  skin: SKIN.general,
};

export default Badge;
