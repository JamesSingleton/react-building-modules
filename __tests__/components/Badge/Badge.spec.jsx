import React from 'react';
import { mount } from 'enzyme';
import Badge from '../../../src/components/Badge';

describe('Badge', () => {
  it('should render without crashing', () => {
    const props = {
      text: 'Hello World',
      onClick: jest.fn(),
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders a small Badge component', () => {
    const props = {
      size: 'small',
      text: 'Hello World',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders a medium Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders an outlined Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders an outlined standard Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
      skin: 'standard',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });

  it('renders an outlined danger Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
      skin: 'danger',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders an outlined success Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
      skin: 'success',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders an outlined warning Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
      skin: 'warning',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
  it('renders an outlined premium Badge component', () => {
    const props = {
      size: 'medium',
      text: 'Hello World',
      type: 'outlined',
      skin: 'premium',
    };
    const renderedBadge = mount(<Badge {...props} />);
    expect(renderedBadge).toMatchSnapshot();
  });
});
