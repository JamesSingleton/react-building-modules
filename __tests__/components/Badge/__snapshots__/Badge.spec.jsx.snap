// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders a medium Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="general"
  text="Hello World"
  type="solid"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="general"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="solid"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "jaOFVS",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="general"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="solid"
    >
      <div
        className="sc-bdVaJa jaOFVS"
        onClick={[Function]}
        size="medium"
        type="solid"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders a small Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="small"
  skin="general"
  text="Hello World"
  type="solid"
>
  <styled.div
    onClick={[Function]}
    size="small"
    skin="general"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="solid"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "htjlZg",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="small"
      skin="general"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="solid"
    >
      <div
        className="sc-bdVaJa htjlZg"
        onClick={[Function]}
        size="small"
        type="solid"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="general"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="general"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "pHZjz",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="general"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa pHZjz"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined danger Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="danger"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="danger"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "AKkzl",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="danger"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa AKkzl"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined premium Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="premium"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="premium"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "gDRSUc",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="premium"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa gDRSUc"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined standard Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="standard"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="standard"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "hhhcd",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="standard"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa hhhcd"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined success Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="success"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="success"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "eBYCJb",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="success"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa eBYCJb"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge renders an outlined warning Badge component 1`] = `
<Badge
  onClick={[Function]}
  size="medium"
  skin="warning"
  text="Hello World"
  type="outlined"
>
  <styled.div
    onClick={[Function]}
    size="medium"
    skin="warning"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="outlined"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fzyIOB",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      size="medium"
      skin="warning"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="outlined"
    >
      <div
        className="sc-bdVaJa fzyIOB"
        onClick={[Function]}
        size="medium"
        type="outlined"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;

exports[`Badge should render without crashing 1`] = `
<Badge
  onClick={[MockFunction]}
  size="medium"
  skin="general"
  text="Hello World"
  type="solid"
>
  <styled.div
    onClick={[MockFunction]}
    size="medium"
    skin="general"
    theme={
      Object {
        "danger": "#F25C5E",
        "general": "#32536A",
        "premium": "#9B61B3",
        "standard": "#006fcf",
        "success": "#43a34c",
        "warning": "#FDB02B",
      }
    }
    type="solid"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "jaOFVS",
            "rules": Array [
              "
  cursor: default;
  border: 1px solid;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  line-height: 1;
  background-color: ",
              [Function],
              ";
  padding: ",
              [Function],
              "
  color: #fff;
  ",
              [Function],
              "
",
            ],
          },
          "defaultProps": Object {
            "theme": Object {
              "danger": "#F25C5E",
              "general": "#32536A",
              "premium": "#9B61B3",
              "standard": "#006fcf",
              "success": "#43a34c",
              "warning": "#FDB02B",
            },
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[MockFunction]}
      size="medium"
      skin="general"
      theme={
        Object {
          "danger": "#F25C5E",
          "general": "#32536A",
          "premium": "#9B61B3",
          "standard": "#006fcf",
          "success": "#43a34c",
          "warning": "#FDB02B",
        }
      }
      type="solid"
    >
      <div
        className="sc-bdVaJa jaOFVS"
        onClick={[MockFunction]}
        size="medium"
        type="solid"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "fySOme",
                  "rules": Array [
                    "
  font-size: 0.625rem;
  line-height: 0.75rem;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-bwzfXH fySOme"
            >
              Hello World
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</Badge>
`;
